conftest.py

```diff
@@ -712,6 +746,29 @@ def setup_test_colo(
         # Check to make sure that limit_db_cpus made it into the colo settings
         return colo_model
 
+
+@pytest.fixture(scope="function")
+def global_dragon_teardown() -> t.Generator[t.Any, t.Any, t.Any]:
+    """Connect to a dragon server started at the path indicated by
+    the environment variable SMARTSIM_DRAGON_SERVER_PATH and
+    force its shutdown to bring down the runtime and allow a subsequent
+    allocation of a new runtime.
+    """
+    if test_launcher != "dragon" or CONFIG.dragon_server_path is None:
+        return
+    exp_path = os.path.join(test_output_root, "dragon_teardown")
+    os.makedirs(exp_path, exist_ok=True)
+    exp: Experiment = Experiment("dragon_shutdown", exp_path=exp_path, launcher=test_launcher)
+    rs = exp.create_run_settings("sleep", ["0.1"])
+    model = exp.create_model("dummy", run_settings=rs)
+    exp.generate(model, overwrite=True)
+    exp.start(model, block=True)
+
+    launcher: DragonLauncher = exp._control._launcher
+    _dragon_cleanup(launcher._dragon_head_socket, launcher._dragon_head_pid)
```

Comment:
Is there a reason why we can't create a wrapper around `_dragon_cleanup` in `DragonLauncher`?  This could avoid using a private method to access private members w/something like `launcher.cleanup()` and it feels like you really want it to be available.

I was about to suggest adding `cleanup` to the base class `Launcher` but it seems we already have `stop`. Do you think we could implement stop in the `DragonLauncher` and replace this line in the test w/`launcher.stop`?

Reply:
OK, I see what you mean -- I think we can add `cleanup()` to `Launcher`. I will do it and ask you to please double check it. `Launcher.stop()` is the function which requests to stop jobs, not to stop the launcher itself, therefore we cannot override it.